ESSIFI-174 Implement the challenge token and session in the REST API

zoemaas opened the pull request:
- Fixed a bug in the POST presentation endpoint;
   - Changed error handling to be spec compliant;
   - Added test cases to from interop;

Commit:
ESSIFI-174 Implement the challenge token and session in the REST API

   - Fixed a bug in the POST presentation endpoint;
   - Changed error handling to be spec compliant;
   - Added test cases to from interop;

Commit:
ESSIFI-174 Implement the challenge token and session in the REST API

   - Extracted environment variables;
   - Updated Dockerfile and docker-compose.yml;
   - Updated the entities;
   - Added test cases;

Commit:
Merge branch 'develop' of github.com:Sphereon-Opensource/pex-ms into feature/ESSIFI-174

 Conflicts:
	docker-compose.yml
	ormconfig.js

hrehman-sphereon reviewed (changes requested):
1. The localhost that is returning in some of the responses should be dynamically changed. For example when we deploy on sphereon.com the response should have sphereon.com in the response not the localhost.
2. The readme can be improved a bit for the postMan testing. And the local debugging. Steps are not executable in their exact current form.
3. As I understand it the challenge token is being used as follows: the challenge token should not be exactly the same as we used for last request. But I think it is not correct. We should accept the challenge token that was returned in the last response. It should be exactly the same ...

We talked over the call to narrow down these points. But if there is need for further discussion on Point 3 please include Niels as well.

Commit:
ESSIFI-174 Implement the challenge token and session in the REST API

   - Updated documentation;
   - Fixed tests;
   - Skip thread.id and challenge.token validation;
   - Fixed error handler;

Commit:
ESSIFI-174 Implement the challenge token and session in the REST API

   - Fixed entity validation;

Commit:
ESSIFI-174 Implement the challenge token and session in the REST API

   - Removed errors from /presentations response body in case the presentation satisfies all input descriptors;

Commit:
ESSIFI-174 Implement the challenge token and session in the REST API

   - Added areRequiredCredentialsPresent: Status to POST /presentations response;
   - Fixed failing tests;
